graph/preprocessor.go

```diff
@@ -292,24 +294,26 @@ func basicAliasSeparation(data []byte) ([]byte, []byte) {
 	return aliasBuffer.Bytes(), buffer.Bytes()
 }
 
-// FindVersion determines the current version of an Alias task file
+// FindVersion determines the current version of a task file
 func FindVersion(data []byte) string {
 	reader := bytes.NewReader(data)
 	scanner := bufio.NewScanner(reader)
 	scanner.Split(bufio.ScanLines)
 	for scanner.Scan() {
 		text := scanner.Text()
-
 		trimmedText := strings.TrimSpace(text)
 
 		// Skip comments and just whitespace.
 		if trimmedText == "" || strings.HasPrefix(trimmedText, "#") {
 			continue
 		}
 
-		// use text instead of trimmedText since '   version: ' is also invalid.
-		if strings.HasPrefix(text, "version:") {
-			return strings.TrimSpace(strings.TrimPrefix(trimmedText, "version:"))
+		// use text instead of trimmedText since ' version: ' is also invalid.
+		if strings.HasPrefix(text, versionKey) {
+			tokens := strings.Split(text, ":")
```

Comment:
What happens if the version is `version: "1.23.4:beta"` which is valid yaml?

Reply:
Discussed offline, but for transparency recommendation is strings.SplitN or strings.SplitAfterN